pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -984,32 +984,57 @@ private void fixColumnName(String rawTableName, TransformExpressionTree expressi
   private void fixColumnName(String rawTableName, Expression expression, @Nullable Map<String, String> columnNameMap) {
     ExpressionType expressionType = expression.getType();
     if (expressionType == ExpressionType.IDENTIFIER) {
-      Identifier identifier = expression.getIdentifier();
-      identifier.setName(getActualColumnName(rawTableName, identifier.getName(), columnNameMap));
+      if (!isStarIdentifier(expression)) {
```

Comment:
This special casing just for emitting a metric does not seem right. Perhaps it might be more straight forward to do it on the server side as follows:

- For each segment, if the segment gets pruned due to missing column, we return that to broker as part of metadata (or executionstatistics). May be we can emit a metric there.

- Broker can decide to act on what to do (metric or metadata in response, or both) if it sees server responses where segments where pruned due to missing column.

cc: @mcvsubbu

Reply:
I agree special casing is not the best.
We don't want to propagate the query to the servers either, knowing that there is a column check that will fail (as it does now).

Reply:
One way we can avoid a bit of code is if we use `SqlShuttle` on the compiled `SqlNode`, to pull out `SqlIdentifier` names, and check if they are in schema. We will still have to exclude the target of `AS`, and in order to do so, will need to walk the `SqlNode` and first recognize all such targets.

I can't think of other methods.

I do think that this is better resolved in the broker than in the server.

@kishoreg , @siddharthteotia  any ideas?